Prevent memory bloat when creating new wxThreads.

nietsu opened the pull request:
The log functions in PthreadStart cause a new wxThreadSpecificInfo object to be created after the original one has already been discarded by CallEntry. Perform the wxThreadSpecificInfo::ThreadCleanUp again at the end of PthreadStart, so that it will not needlessly remain there consuming memory till the end of program.

nietsu committed:
Prevent memory bloat when creating new wxThreads.

The log functions in PthreadStart cause a new wxThreadSpecificInfo object to be created after the original one has already been discarded by CallEntry.
Perform the wxThreadSpecificInfo::ThreadCleanUp again at the end of PthreadStart, so that it will not needlessly remain there consuming memory till the end of program.

vadz reviewed (commented):
Thanks for finding and fixing this problem!

I wonder if we should stop calling `ThreadCleanUp()` from `CallEntry()` (and then just remove it completely, as it won't be doing anything useful then) to avoid calling it twice?

nietsu commented:
> I wonder if we should stop calling `ThreadCleanUp()` from `CallEntry()` (and then just remove it completely, as it won't be doing anything useful then) to avoid calling it twice?

I agree it's a bit messy to call it twice. Still, I think at least Window implementation still relies on the call from CallEntry(), so you probably shouldn't just remove it without creating some other method of calling it on platforms that don't use the implementation in threadpsx.cpp. Just calling it directly from each implementation would be a solution, although a bit fragile one.

vadz committed:
Drop wxThread::CallEntry() and clean up in platform-specific code

Calling wxThreadSpecificInfo::ThreadCleanUp() in CallEntry() was
premature as thread-specific info could be recreated later, e.g. due to
any wxLogXXX() call, so don't do it there, but do it in the
platform-specific code just before exiting the thread.

As this renders CallEntry() useless, remove it entirely.

vadz commented:
I've added a commit to do this, could you please check if it works correctly for you? TIA!

nietsu commented:
Just tested it. Works great!
